crates/ruff_cli/src/lib.rs

```diff
@@ -1,27 +1,288 @@
-//! This library only exists to enable the Ruff internal tooling (`ruff_dev`)
-//! to automatically update the `ruff help` output in the `README.md`.
-//!
-//! For the actual Ruff library, see [`ruff`].
+use anyhow::Result;
+use clap::CommandFactory;
+use notify::{recommended_watcher, RecursiveMode, Watcher};
+use std::io;
+use std::path::PathBuf;
+use std::process::ExitCode;
+use std::sync::mpsc::channel;
 
-mod args;
+use crate::args::{Args, CheckArgs, Command};
+use crate::printer::{Flags as PrinterFlags, Printer};
 
-use clap::CommandFactory;
+use ruff::logging::{set_up_logging, LogLevel};
+use ruff::settings::types::SerializationFormat;
+use ruff::settings::CliSettings;
+use ruff::{fix, fs, warn_user_once};
+
+pub mod args;
+mod cache;
+mod commands;
+mod diagnostics;
+mod iterators;
+mod printer;
+mod resolve;
+
+pub enum ExitStatus {
+    /// Linting was successful and there were no linting errors.
+    Success,
+    /// Linting was successful but there were linting errors.
+    Failure,
+    /// Linting failed.
+    Error,
+}
+
+impl From<ExitStatus> for ExitCode {
+    fn from(status: ExitStatus) -> Self {
+        match status {
+            ExitStatus::Success => ExitCode::from(0),
+            ExitStatus::Failure => ExitCode::from(1),
+            ExitStatus::Error => ExitCode::from(2),
+        }
+    }
+}
 
-/// Returns the output of `ruff help`.
-pub fn command_help() -> String {
-    args::Args::command().render_help().to_string()
+pub fn run(
```

Comment:
Most of it is moved from `main.rs`. The main difference is that it now exposes a `run` function with the arguments.